includes/api/wc-blocks/class-wc-blocks-rest-product-attribute-terms-controller.php

```diff
@@ -0,0 +1,169 @@
+					'args'                => array(
+						'context' => $this->get_context_param(
+							array(
+								'default' => 'view',
+							)
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check permissions.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @param string          $context Request context.
+	 * @return bool|WP_Error
+	 */
+	protected function check_permissions( $request, $context = 'read' ) {
+		// Get taxonomy.
+		$taxonomy = $this->get_taxonomy( $request );
+		if ( ! $taxonomy || ! taxonomy_exists( $taxonomy ) ) {
+			return new WP_Error( 'woocommerce_rest_taxonomy_invalid', __( 'Taxonomy does not exist.', 'woo-gutenberg-products-block' ), array( 'status' => 404 ) );
+		}
+
+		// Check permissions for a single term.
+		$id = intval( $request['id'] );
+		if ( $id ) {
+			$term = get_term( $id, $taxonomy );
+
+			if ( is_wp_error( $term ) || ! $term || $term->taxonomy !== $taxonomy ) {
+				return new WP_Error( 'woocommerce_rest_term_invalid', __( 'Resource does not exist.', 'woo-gutenberg-products-block' ), array( 'status' => 404 ) );
+			}
+		}
+
+		return current_user_can( 'edit_posts' );
+	}
+
+	/**
+	 * Prepare a single product category output for response.
+	 *
+	 * @param WP_Term         $item    Term object.
+	 * @param WP_REST_Request $request Request instance.
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		// Get the attribute slug.
+		$attribute_id = absint( $request->get_param( 'attribute_id' ) );
+		$attribute    = wc_get_attribute( $attribute_id );
+
+		$data = array(
+			'id'        => (int) $item->term_id,
+			'name'      => $item->name,
+			'slug'      => $item->slug,
+			'count'     => (int) $item->count,
+			'attr_name' => $attribute->name,
+			'attr_slug' => $attribute->slug,
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $item, $request ) );
+
+		return $response;
+	}
+
+	/**
+	 * Get the Product's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$raw_schema = parent::get_item_schema();
+		$schema     = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'product_block_category',
+			'type'       => 'object',
+			'properties' => array(),
+		);
+
+		$schema['properties']['id']        = $raw_schema['properties']['id'];
+		$schema['properties']['name']      = $raw_schema['properties']['name'];
+		$schema['properties']['slug']      = $raw_schema['properties']['slug'];
+		$schema['properties']['count']     = $raw_schema['properties']['count'];
+		$schema['properties']['attr_name'] = array(
```

Comment:
I think we could keep with my notes from the last PR:
https://github.com/woocommerce/woocommerce/pull/22793#discussion_r258789961

So we don't can keep all data from attributes grouped, since it's come from a parent resource.

Reply:
I updated this, though just a tad bit different - the schema you provided made it look like an array of attributes here possibly, let me know if what I pushed in 283d7f4